### PR TITLE
docs : Add CONTRIBUTING.md for project contribution guidelines

### PR DESCRIPTION
**Pull request by sephali-osm**
Add CONTRIBUTING.md for project contribution guidelines

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# Contributing to document-service
```
Comment (sameer-s-b) [inline CONTRIBUTING.md:1]: We follow a specific convention for headings.

Only the first word starts with CAPS unless the other word is a noun.
Refer the README.md file in this project to understand how the headings conventions is followed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+### Submitting a Pull Request (PR)
+
+Before submitting a Pull Request (PR), please follow these guidelines:
+
+1. Search GitHub [Pull Requests](https://github.com/OsmosysSoftware/document-service/pulls) to ensure there is no open or closed PR 
+   related to your submission.
+2. Fork this repository.
+3. Make your changes in a new Git branch.
+   ```csharp
```
Comment (sameer-s-b) [inline CONTRIBUTING.md:54]: This is a command to be ran in CMD. The formatting need not be in csharp. Remove this.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+   git checkout -b my-fix-branch main
+   ```
+4. Create your patch, **including appropriate test cases**.
+5. Follow our coding rules, which is mentioned below.
+6. Run the full document-service test suite to ensure that all tests pass.
```
Comment (sameer-s-b) [inline CONTRIBUTING.md:59]: We don't have a test suite yet. Nor do we run any test cases before we submit PRs. These points number 4 and number 6 can be removed.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+   Note: the optional commit -a command line option will automatically "add" and "rm" edited files.
+8. Push your branch to your GitHub repository.
+   ```csharp
+   git push origin my-fix-branch
+   ```
+9. Send a pull request to the `document-service:main`.
+
+- <strong style="color:black">�</strong> **If we suggest changes, then:**
+  - � Make the required updates.
+  - � Re-run the Osmo-Notify test suites to ensure tests are still passing.
```
Comment (sameer-s-b) [inline CONTRIBUTING.md:73]: Osmo-Notify --> document-service

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+an issue. Major features require discussion, so please prefix your proposal with [discussion], for example, "[discussion]: your feature idea.
+" Smaller features can be crafted and directly submitted as a Pull Request.
+
+## Coding Rules
+
+To ensure consistency throughout the source code, follow these rules as you work on the project:
+
+- All features or bug fixes must be tested by one or more specs (unit-tests).
+
+- We follow the [visual studio coding standards](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions),
```
Comment (sameer-s-b) [inline CONTRIBUTING.md:90]: You can link to the Omsosys coding standards for .NET/C#
https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/dotnet.md

**Review by sameer-s-b — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (sephali-osm) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (sephali-osm) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 3 above)
Comment (sephali-osm) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 4 above)
Comment (sephali-osm) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 5 above)
Comment (sephali-osm) [reply CONTRIBUTING.md]: Updated

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**